File: java/IEC61970/InfIEC61970/InfOperationalLimits/TemperaturePolynomialLimit.java
package IEC61970.InfIEC61970.InfOperationalLimits;

import IEC61970.Base.Domain.Float;

/**
 * This describes the coefficients of a polynomial function that has temperature
 * as input and calculates limit values as output.
 * @author akamath
 * @version 1.0
 * @created 25-Dec-2023 8:32:04 PM
 */
public class TemperaturePolynomialLimit extends EnvironmentalDependentLimit {

	/**
	 * The polinomial coefficent of power 0.
	 */
	public Float coefficient0;
	/**
	 * The polinomial coefficent of power 1.
	 */
	public Float coefficient1;
	/**
	 * The polinomial coefficent of power 2.
	 */
	public Float coefficient2;
	/**
	 * The polinomial coefficent of power 3.
	 */
	public Float coefficient3;
	/**
	 * The polinomial coefficent of power 4.
	 */
	public Float coefficient4;

	public TemperaturePolynomialLimit(){

	}
}//end TemperaturePolynomialLimit

File: py/IEC61970/InfIEC61970/InfOperationalLimits/TemperaturePolynomialLimit.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106 on Wed Jan  3 22:25:18 2024
from IEC61970.InfIEC61970.InfOperationalLimits.EnvironmentalDependentLimit import EnvironmentalDependentLimit


class TemperaturePolynomialLimit(EnvironmentalDependentLimit):
    """
    Represents the coefficients of a polynomial function that has temperature as input and calculates limit values as output.
    """
    def __init__(self):
        super().__init__()
        self.coefficient0 = 0.0  # The polynomial coefficient of power 0.
        self.coefficient1 = 0.0  # The polynomial coefficient of power 1.
        self.coefficient2 = 0.0  # The polynomial coefficient of power 2.
        self.coefficient3 = 0.0  # The polynomial coefficient of power 3.
        self.coefficient4 = 0.0  # The polynomial coefficient of power 4.


